jenkins_jobs/trigger_host_os_iso_build_from_versions_repo/script.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+set -x
+
+
+TARGET_CONTEXT="Build Host OS"
+
+
+get_build_state(){
+    # this will evaluate the queried keys as variables: state=<state>,
+    # target_url=<url>
+    eval $(github_api "$GITHUB_USER_NAME" "$GITHUB_PASSWORD" \
+                      query_status "${GITHUB_ORGANIZATION_NAME}/versions" \
+                      "$ghprbPullId" "$TARGET_CONTEXT" --state \
+                      --target-url || echo "exit 1")
+}
+
+get_pr_state(){
+    # this will evaluate the queried keys as variables: state=<state>,
+    # title=<title>
+    eval $(github_api "$GITHUB_USER_NAME" "$GITHUB_PASSWORD" \
+                      query_pr "${GITHUB_ORGANIZATION_NAME}/versions" \
+                      "$ghprbPullId" --state --title || echo "exit 1")
+}
+
+state="pending"
+get_build_state
```

Comment:
I just remembered you cannot return a string in shell script. I really dislike shell script!

http://stackoverflow.com/questions/3236871/how-to-return-a-string-value-from-a-bash-function
2nd answer suggests passing a variable "by-reference" and altering it inside the function, it seems clearer to me, at least you are altering a function parameter and not *any* global script variable you choose.